geode-core/src/main/java/org/apache/geode/management/internal/cli/commands/DestroyGatewayReceiverCommand.java

```diff
@@ -57,28 +61,37 @@ public Result destroyGatewaySender(
       @CliOption(key = CliStrings.IFEXISTS, help = CliStrings.IFEXISTS_HELP,
           specifiedDefaultValue = "true", unspecifiedDefaultValue = "false") boolean ifExists) {
 
+    boolean persisted = false;
     Set<DistributedMember> members = getMembers(onGroups, onMember);
 
     List<CliFunctionResult> functionResults =
         executeAndGetFunctionResult(DestroyGatewayReceiverFunction.INSTANCE, ifExists, members);
 
     CommandResult result = ResultBuilder.buildResult(functionResults);
-    XmlEntity xmlEntity = findXmlEntity(functionResults);
 
-    // no xml needs to be updated, simply return
-    if (xmlEntity == null) {
-      return result;
-    }
+    // Only update the cluster config if the command is not executed on specific members.
+    InternalClusterConfigurationService service =
+        (InternalClusterConfigurationService) getConfigurationService();
+    if (onMember == null && result.getStatus().equals(Result.Status.OK) && service != null) {
+      List<String> groups = Arrays.asList(Optional.ofNullable(onGroups).orElse(new String[0]));
```

Comment:
this is probably easier to read:
    if(onGroups == null){
      onGroups = new String[]{"cluster"};
    }
    Arrays.stream(onGroups).forEach(group->{});